crates/object-store/src/file.rs

```diff
@@ -25,9 +22,29 @@ impl FileObjectStore {
     }
 }
 
-impl From<FileObjectStore> for FileObjectStoreData {
-    fn from(fos: FileObjectStore) -> Self {
-        (fos, Default::default())
+pub struct FileObjectStoreComponent {
+    pub root: Path,
+}
+
+impl HostComponent for FileObjectStoreComponent {
+    type Data = (
+        FileObjectStore,
+        spin_object_store::SpinObjectStoreTables<FileObjectStore>,
+    );
+
+    fn add_to_linker<T>(
+        linker: &mut wasmtime::Linker<spin_engine::RuntimeContext<T>>,
+        data_handle: HostComponentsDataHandle<Self::Data>,
+    ) -> anyhow::Result<()> {
+        crate::add_to_linker(linker, move |ctx| {
+            let (data, table) = data_handle.get_mut(ctx);
+            (data, table)
```

Comment:
Given this deconstructs and immediately reconstructs the tuple, why not just `move |ctx| data_handle.get_mut(ctx)`?

Reply:
`get_mut` returns e.g. `&mut (A, B)`, but this closure needs (&mut A, &mut B). I also spent an unreasonable amount of time trying to fix this with no luck. Suggests welcome. If they work.

Reply:
Aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaah.  That's cool - I just completely missed that nuance.